File: Back/WebShop-Backend/src/main/java/com/example/WebShop/services/UserService.java
package com.example.WebShop.services;

import com.example.WebShop.models.dto.User;
import com.example.WebShop.models.requests.ActivateAccountRequest;
import com.example.WebShop.models.requests.SignUpRequest;
import com.example.WebShop.models.requests.UserUpdateRequest;
import org.springframework.security.core.Authentication;

public interface UserService {
    User updateUser(String username, UserUpdateRequest user, Authentication auth);
    User findById(Integer id);
    void signUp(SignUpRequest request);
    User activateAccount(String username);
}

File: Front/WebShop-Frontend/src/app/services/user.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { TokenService } from '../auth/services/token.service';
import { UserUpdateRequest } from '../models/userUpdateRequest';
import { User } from '../models/user';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  private baseUrl = 'http://localhost:9000/users';

  constructor(private http: HttpClient, private tokenService: TokenService) { }

  updateUser(request: UserUpdateRequest){
    const headers = new HttpHeaders({
      Authorization: `Bearer ${this.tokenService.getAccessToken()}`
    });
    
    return this.http.put<User>(`${this.baseUrl}/` + this.tokenService.getUser().username, request, { headers })
  }
}
